core/lib/zksync_core/src/l1_gas_price/singleton.rs

```diff
@@ -59,17 +49,23 @@ impl GasAdjusterSingleton {
                 .context("GasAdjuster::new()")?;
                 Ok(Arc::new(adjuster))
             })
-            .await;
-        adjuster.clone()
+            .await
+        {
+            Ok(adjuster) => Ok(adjuster.clone()),
+            Err(_e) => Err(anyhow::anyhow!("Failed to get or initialize GasAdjuster")),
+        }
```

Comment:
Can't you use `?` in here?

Reply:
Replacing the `match` with the `?` operator results in the following compilation error:
```
the `?` operator can only be applied to values that implement `Try`
the trait `Try` is not implemented for `&std::result::Result<Arc<GasAdjuster<QueryClient>>, anyhow::Error>`
the trait `Try` is implemented for `std::result::Result<T, E>`
```

The output of `get_or_init()` is not a `Result`, but a reference (`&`) to it. The simplest solution would be to `.clone()` the reference to get it's inner value, but this is not possible, because `anyhow::Error` is not cloneable, so when calling `clone()` on it, instead of getting the contents of the reference, we get the same reference again.

Let me know if I explained something poorly.